test/e2e/secrets_test.go

```diff
@@ -181,9 +181,9 @@ func testPrivateKeyChange(t *testing.T) {
 	err = tc.validateUpgradeableCondition(meta.ConditionTrue)
 	require.NoError(t, err, "operator Upgradeable condition not in proper state")
 
-	// Re-create the known private key so SSH connection can be re-established
-	// TODO: Remove dependency on this secret by rotating keys as part of https://issues.redhat.com/browse/WINC-655
-	require.NoError(t, tc.createPrivateKeySecret(true), "error confirming known private key secret exists")
+	// revert key changes so the test suite is able to SSH into the VMs
+	require.NoError(t, tc.createPrivateKeySecret(true))
+	require.NoError(t, tc.waitForNewMachineNodes())
```

Comment:
Do we need to always do this almost like a defer function call? i.e. will we have issues collecting node logs upon one of the require failures above?

Reply:
It should only be done directly after the private-key change, calling it at another time would throw an error.

Reply:
Correct, so do we need to call always even if:
```
	err = tc.validateUpgradeableCondition(meta.ConditionTrue)
	require.NoError(t, err, "operator Upgradeable condition not in proper state")
```
fails?

Reply:
Nope, that is before the private key change here.

If you're asking about the private key change on line 168, I would say that thats not an issue I'm trying to solve in this PR. This is fixing the CI issue where the reconfigure test returns green, and then CI continues onto the deletion test.

In the case that specific validateUpgradeableCondition call fails, the reconfigure test will return an error and the deletion test will not be called. I am not very concerned about the node taking some time to return to a working state after that.

Reply:
> the reconfigure test will return an error and the deletion test will not be called. 

And we don't try to collect logs at that point by SSHing into nodes? If so, I am fine with the change.